tests/countries/test_saint_helena_ascension_and_tristan_da_cunha.py

```diff
@@ -0,0 +1,273 @@
+        )
+        obs_dt = (
+            "2017-01-02",
+            "2022-01-03",
+            "2023-01-02",
+        )
+        self.assertHolidayName(f"{name} (observed)", self.government_holidays, obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        self.assertHolidayName(
+            name,
+            "2020-04-10",
+            "2021-04-02",
+            "2022-04-15",
+            "2023-04-07",
+            "2024-03-29",
+            "2025-04-18",
+        )
+        self.assertHolidayName(name, range(2015, 2050))
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+        self.assertHolidayName(
+            name,
+            "2020-04-13",
+            "2021-04-05",
+            "2022-04-18",
+            "2023-04-10",
+            "2024-04-01",
+            "2025-04-21",
+        )
+        self.assertHolidayName(name, range(2015, 2050))
+
+    def test_queens_birthday(self):
+        name = "Queen's Birthday"
+        self.assertHolidayName(
+            name,
+            "2018-06-18",
+            "2019-06-17",
+            "2020-06-08",
+            "2021-06-14",
+            "2022-06-13",
+        )
+        self.assertHolidayName(name, range(2015, 2023))
+        self.assertNoHolidayName(name, range(2023, 2050))
+
+    def test_kings_birthday(self):
+        name = "King's Birthday"
+        self.assertHolidayName(
+            name,
+            "2023-06-19",
+            "2024-11-15",
+            "2025-11-14",
+        )
+        self.assertHolidayName(name, range(2023, 2050))
+        self.assertNoHolidayName(name, range(2015, 2023))
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertHolidayName(name, (f"{year}-12-25" for year in range(2015, 2050)))
+        obs_dt = (
+            "2016-12-27",
+            "2021-12-27",
+            "2022-12-27",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_boxing_day(self):
+        name = "Boxing Day"
+        self.assertHolidayName(name, (f"{year}-12-26" for year in range(2015, 2050)))
+        obs_dt = (
+            "2015-12-28",
+            "2020-12-28",
+            "2021-12-28",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_whit_monday(self):
+        name = "Whit Monday"
+        self.assertNoHolidayName(name)
+        self.assertHolidayName(
+            name,
+            self.government_holidays,
+            "2020-06-01",
+            "2021-05-24",
+            "2022-06-06",
+            "2023-05-29",
+            "2024-05-20",
+            "2025-06-09",
+        )
+        self.assertHolidayName(name, self.government_holidays, range(2015, 2050))
+
+    def test_subdivision_public_holidays(self):
```

Comment:
There should be separate methods: `test_ascension_day`, `test_saint_helena_day` etc.